utils/entity.js

```diff
@@ -191,57 +186,46 @@ function prepareEntityForTemplates(entityWithConfig, generator) {
-              derivedEntity: relationship.otherEntity,
-              jpaGeneratedValue: false,
-              liquibaseAutoIncrement: false,
-              // Mapsid is generated by relationship select
-              autoGenerate: true,
-              readonly: true,
-              get fieldName() {
-                return idCount === 1 ? field.fieldName : `${relationship.otherEntity.entityInstance}${field.fieldNameCapitalized}`;
-              },
-              get fieldNameCapitalized() {
-                return idCount === 1 ? field.fieldNameCapitalized : `${relationship.otherEntity.entityClass}${field.fieldNameCapitalized}`;
-              },
-              get columnName() {
-                return idCount === 1 ? field.columnName : `${generator.getColumnName(relationship.relationshipName)}_${field.columnName}`;
-              },
-              get reference() {
-                return fieldToReference(entityWithConfig, this);
-              },
-            }));
-          },
-        };
       });
     }
 
     if (idCount === 1 && idRelationships.length === 1) {
       const relationshipId = idRelationships[0];
       // One-To-One relationships with id uses @MapsId.
       // Almost every info is taken from the parent, except some info like autoGenerate and derived.
-      // calling fieldName as id is for backward compatibility, in the future we may want to prefix it with relationship name.
+      // calling fieldName as otherEntityIdField.fieldName is for backward compatibility, in the future we may want to prefix it with relationship name.
+      entityWithConfig._fields = entityWithConfig.fields;
+      if (!relationshipId.otherEntity.primaryKey.composite) {
+        Object.defineProperty(entityWithConfig, 'fields', {
+          get: () => {
+            if (!entityWithConfig.idField) {
+              const otherEntityIdField = relationshipId.otherEntity.primaryKey.originalFields[0];
+              entityWithConfig.idField = {
+                fieldName: otherEntityIdField.fieldName,
+                fieldType: otherEntityIdField.fieldType,
+                id: true,
+                autoGenerate: true,
+                jpaGeneratedValue: false,
+                liquibaseAutoIncrement: false,
+                derived: true,
+                derivedEntity: relationshipId.otherEntity,
+                readonly: true,
+              };
+            }
+            return [entityWithConfig.idField, ...entityWithConfig._fields];
+          },
+        });
+      }
       entityWithConfig.primaryKey = {
-        fieldName: 'id',
         derived: true,
         // MapsId copy the id from the relationship.
         autoGenerate: true,
-        get fields() {
-          return this.derivedFields;
+        get ids() {
+          return relationshipToIds(relationshipId, generator);
         },
-        get derivedFields() {
-          return relationshipId.derivedPrimaryKey.derivedFields;
+        get originalFields() {
```

Comment:
Really sorry, but originalFields is not spelling at all: what does it mean?
Isn't something like 'composingFields' would be more understandable

Reply:
@Tcharl thanks for the feedback, I myself couldn't find a better name but maybe you could help. I will go with what the majority decides.

to answer your question a primaryKey boils down to the list of columns in the database that constitute the primary key of its entity. The column they refer to might be few entities away, originalFields represents the list of field in the original entities where they where first defined.

Ex: 
let's define the following JDL:
```

entity Employee{
	@id username String required
	fullname String required
}

entity EmployeeSkill{
	// compositeId employeeSkill.name, employee.id
	@id name String required
	level Integer required
}

relationship OneToMany{
	@id Employee{skill(name)} to EmployeeSkill{employee(fullname) required}
}

entity CertificateType{
	name String required
}

entity EmployeeSkillCertificate{
    // EmployeeSkill.employee.username, EmployeeSkill.name, CertficiateType.id
    grade Integer required
    date LocalDate required
}

relationship OneToMany{
	@id CertificateType to EmployeeSkillCertificate{type(name) required}
	@id EmployeeSkill to EmployeeSkillCertificate{skill required}
}
```

EmployeeSkillCertificate has primaryKey that contains 3 `ids`:
EmployeeSkill.employee.username, EmployeeSkill.name, CertficiateType.id

originalFields are:
- username (in the entity employee)
- name (in the entity employeeSkill)
- id (in the entity certificate type)

I do not personally like  `composingFields` since they are not really in the primaryKey and composing it, but if you agree on it I will use it.

Reply:
more clear now, I thought that it was a deep copy instead of an alias.
Maybe being even more specific, like `referencedIdsComposingId`, WDYT?

Reply:
I would prefer not to use reference in the name, since it's refers to something else in the code as you might know (kind of in intersection between field and relationship transformed) a good name would be <SOMETHING>Fields since they are fields and when you use then you know they have fieldName/type... but again we could change these later